code/game/gamemodes/miniantags/mosquito/mosquito_event.dm

```diff
@@ -0,0 +1,69 @@
+/datum/event/spawn_mosquito
+	var/SuccesAnnouncement = FALSE
+	var/successSpawn = FALSE
+	var/spawncount = 5
+	var/datum/disease/ChosenDisease
+
+/datum/event/spawn_mosquito/setup()
+	spawncount = rand(3, 6)
+
+/datum/event/spawn_mosquito/start()
+	INVOKE_ASYNC(src, .proc/spawn_mosquitos)
+
+/datum/event/spawn_mosquito/proc/spawn_mosquitos()
+	var/list/possible_diseases = list(
+	/datum/disease/cold,
+	/datum/disease/flu,
+	/datum/disease/magnitis,
+	/datum/disease/appendicitis,
+	/datum/disease/cold9)
+	ChosenDisease = pick(possible_diseases)
+
+	var/list/vents = get_valid_vent_spawns(exclude_mobs_nearby = TRUE, exclude_visible_by_mobs = TRUE) // gets all vent locations
+	var/list/pickedcandidates = new()
+	var/list/candidates = SSghost_spawns.poll_candidates("Do you want to play as a mosquito?", ROLE_MOSQUITO, TRUE, source = /mob/living/simple_animal/mosquito) // gets ghosts
+
+	if(!length(candidates)) // if there are no picked candidates
+		kill()
+		message_admins("No candidates for mosquito event")
+		return
+
+	while(spawncount && length(candidates)) // should be for forloop
```

Comment:
if you want a comment to be a reminder for you to do something then start it with `// todo` - most modern code editors highlight it for you and might even remind you about it before a commit

if you want your comment to instead be a piece of documentation for the next person reading your code nine times out of ten you can probably just write cleaner code

All of the comments in this file can be removed.

Reply:
Adjusted accordingly